source/routing/raptor_api.cpp

```diff
@@ -1058,97 +1051,131 @@ void make_response(navitia::PbCreator& pb_creator,
-        return;
     }
-
-    const auto direct_path = get_direct_path(worker, origin, destination);
-
-    if(departures && (departures->size() == 0) && destinations && (destinations->size() == 0)){
-        make_pathes(pb_creator, pathes, worker, direct_path, origin, destination, datetimes, clockwise);
-        if (pb_creator.has_response_type(pbnavitia::NO_SOLUTION)) {
+    // case 3 : departure and destination is emtpy
+    else if (departures && (departures->size() == 0) && destinations && (destinations->size() == 0)){
+        make_pathes(pb_creator,
+                    pathes,
+                    worker,
+                    get_direct_path(worker, origin, destination),
+                    origin,
+                    destination,
+                    datetimes,
+                    clockwise);
+        if (pb_creator.empty_journeys()) {
             pb_creator.fill_pb_error(pbnavitia::Error::no_origin_nor_destination,
                                      pbnavitia::NO_ORIGIN_NOR_DESTINATION_POINT,
                                      "no origin point nor destination point");
         }
-        return;
     }
-
-    if(departures && departures->size() == 0){
-        make_pathes(pb_creator, pathes, worker, direct_path, origin, destination, datetimes, clockwise);
-        if (pb_creator.has_response_type(pbnavitia::NO_SOLUTION)) {
+    // case 4 : only departure is emtpy
+    else if (departures && departures->size() == 0){
+        make_pathes(pb_creator,
+                    pathes,
+                    worker,
+                    get_direct_path(worker, origin, destination),
+                    origin,
+                    destination,
+                    datetimes,
+                    clockwise);
+        if (pb_creator.empty_journeys()) {
             pb_creator.fill_pb_error(pbnavitia::Error::no_origin,
                                      pbnavitia::NO_ORIGIN_POINT, "no origin point");
         }
-        return;
     }
-
-    if(destinations && destinations->size() == 0){
-        make_pathes(pb_creator, pathes, worker, direct_path, origin, destination, datetimes, clockwise);
-        if (pb_creator.has_response_type(pbnavitia::NO_SOLUTION)) {
+    // case 5 : only destination is emtpy
+    else if (destinations && destinations->size() == 0){
+        make_pathes(pb_creator,
+                    pathes,
+                    worker,
+                    get_direct_path(worker, origin, destination),
+                    origin,
+                    destination,
+                    datetimes,
+                    clockwise);
+        if (pb_creator.empty_journeys()) {
             pb_creator.fill_pb_error(pbnavitia::Error::no_destination,
                                      pbnavitia::NO_DESTINATION_POINT,
                                      "no destination point");
         }
-        return;
     }
+    // classical case : We have departures and destinations
+    else {
```

Comment:
I don't like having a `else` that isn't on the same line than the closing bracket